File: app/src/main/java/ai/elimu/chat/dao/converter/CalendarConverter.java
package ai.elimu.chat.dao.converter;

import org.greenrobot.greendao.converter.PropertyConverter;

import java.util.Calendar;

public class CalendarConverter implements PropertyConverter<Calendar, Long> {

    @Override
    public Calendar convertToEntityProperty(Long databaseValue) {
        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(databaseValue);
        return calendar;
    }

    @Override
    public Long convertToDatabaseValue(Calendar entityProperty) {
        Long databaseValue = entityProperty.getTimeInMillis();
        return databaseValue;
    }
}

File: app/src/main/java/ai/elimu/chat/dao/converter/CalendarConverter.kt
package ai.elimu.chat.dao.converter

import org.greenrobot.greendao.converter.PropertyConverter
import java.util.Calendar

class CalendarConverter : PropertyConverter<Calendar?, Long?> {

    override fun convertToEntityProperty(databaseValue: Long?): Calendar? {
        val calendar = Calendar.getInstance()
        databaseValue?.let {
            calendar.setTimeInMillis(databaseValue)
        }
        return calendar
    }

    override fun convertToDatabaseValue(entityProperty: Calendar?): Long? {
        val databaseValue = entityProperty?.getTimeInMillis()
        return databaseValue
    }
}
